File: stack_using_queue.py
from collections import deque
class MyStack:
    def __init__(self):
        self.q1 = deque()
        self.q2 = deque()

    def push(self, x: int):
        self.q2.append(x)
        while self.q1:
            self.q2.append(self.q1.popleft())
        self.q1, self.q2 = self.q2, self.q1

    def pop(self):
        return self.q1.popleft()

    def top(self):
        return self.q1[0]

    def empty(self):
        return len(self.q1) == 0

File: Stack Using Queue.java
class MyStack {
    Queue<Integer> que1;
     Queue<Integer> que2;
     int top;

    public MyStack() {
        que1 = new LinkedList<>();
        que2 = new LinkedList<>();
        
        
    }
    
    public void push(int x) {
        que1.offer(x);
        top=x;
        
    }
    
    public int pop() {
        while(que1.size()>1){
            top=que1.poll();
            que2.offer(top);
        }
        int val = que1.poll();
        while(!que2.isEmpty()){
            que1.offer(que2.poll());
        }
        return val;
        
    }
    
    public int top() {
        return top;
        
    }
    
    public boolean empty() {
        return que1.isEmpty();
    }
}
